Sources/MapboxDirectionsCLI/ProcessCommand.swift

```diff
@@ -53,6 +56,29 @@ class ProcessCommand<ResponceType : Codable, OptionsType : DirectionsOptions > :
                let jsonData = try? JSONSerialization.data(withJSONObject: object, options: [.prettyPrinted]) {
                 outputText = String(data: jsonData, encoding: .utf8)!
             }
+        case .gpx:
+            var gpxText: String = String("<?xml version=\"1.0\" encoding=\"UTF-8\"?>")
+            gpxText.append("\n<gpx xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://www.topografix.com/GPX/1/1\" xsi:schemaLocation=\"http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd\" version=\"1.1\">")
+            
+            guard let routeResponse = routeResponse,
+                  let routes = routeResponse.routes else { return }
+            
+            let dateFormatter = ISO8601DateFormatter()
+            dateFormatter.formatOptions = .withInternetDateTime
+            var time = Date()
+            
+            routes.forEach { route in
+                let shape = route.shape
+                let timeInterval = TimeInterval(route.distance/route.expectedTravelTime)
+                for coord in shape!.coordinates {
```

Comment:
I think you shall take into account the distance between coordinates to generate correct timestamp so that the speed will be ok.

Reply:
The template has said this for a few years, but every time I’ve tested it, you can’t get Xcode to interpolate anything.